test/e2e/apimachinery/crd_conversion_webhook.go

```diff
@@ -122,80 +124,88 @@ var _ = SIGDescribe("CustomResourceConversionWebhook", func() {
-
-		deployCustomResourceWebhookAndService(f, imageutils.GetE2EImage(imageutils.Agnhost), context)
 	})
 
 	ginkgo.AfterEach(func() {
 		cleanCRDWebhookTest(client, namespaceName)
 	})
 
-	ginkgo.It("Should be able to convert from CR v1 to CR v2", func() {
-		testcrd, err := crd.CreateMultiVersionTestCRD(f, "stable.example.com", func(crd *apiextensionsv1.CustomResourceDefinition) {
-			crd.Spec.Versions = apiVersions
-			crd.Spec.Conversion = &apiextensionsv1.CustomResourceConversion{
-				Strategy: apiextensionsv1.WebhookConverter,
-				Webhook: &apiextensionsv1.WebhookConversion{
-					ClientConfig: &apiextensionsv1.WebhookClientConfig{
-						CABundle: context.signingCert,
-						Service: &apiextensionsv1.ServiceReference{
-							Namespace: f.Namespace.Name,
-							Name:      serviceCRDName,
-							Path:      pointer.StringPtr("/crdconvert"),
-							Port:      pointer.Int32Ptr(serviceCRDPort),
-						},
-					},
-					ConversionReviewVersions: []string{"v1", "v1beta1"},
-				},
+	ginkgo.Context("Running a conversion webhook service on the default port", func() {
+		ginkgo.BeforeEach(func() {
+			deployCustomResourceWebhookAndService(f, imageutils.GetE2EImage(imageutils.Agnhost), context, servicePort)
+		})
+
+		ginkgo.It("Should be able to convert from CR v1 to CR v2", func() {
+			testcrd, err := createMultiVersionTestCRDWithConversionWebhook(f, context, servicePort)
+			if err != nil {
+				return
 			}
-			crd.Spec.PreserveUnknownFields = false
+			defer testcrd.CleanUp()
+			testCustomResourceConversionWebhook(f, testcrd.Crd, testcrd.DynamicClients)
+		})
+
+		ginkgo.It("Should be able to convert a non homogeneous list of CRs", func() {
+			testcrd, err := createMultiVersionTestCRDWithConversionWebhook(f, context, servicePort)
+			if err != nil {
+				return
+			}
+			defer testcrd.CleanUp()
+			testCRListConversion(f, testcrd)
 		})
-		if err != nil {
-			return
-		}
-		defer testcrd.CleanUp()
-		testCustomResourceConversionWebhook(f, testcrd.Crd, testcrd.DynamicClients)
 	})
 
-	ginkgo.It("Should be able to convert a non homogeneous list of CRs", func() {
-		testcrd, err := crd.CreateMultiVersionTestCRD(f, "stable.example.com", func(crd *apiextensionsv1.CustomResourceDefinition) {
-			crd.Spec.Versions = apiVersions
-			crd.Spec.Conversion = &apiextensionsv1.CustomResourceConversion{
-				Strategy: apiextensionsv1.WebhookConverter,
-				Webhook: &apiextensionsv1.WebhookConversion{
-					ClientConfig: &apiextensionsv1.WebhookClientConfig{
-						CABundle: context.signingCert,
-						Service: &apiextensionsv1.ServiceReference{
-							Namespace: f.Namespace.Name,
-							Name:      serviceCRDName,
-							Path:      pointer.StringPtr("/crdconvert"),
-							Port:      pointer.Int32Ptr(serviceCRDPort),
-						},
-					},
-					ConversionReviewVersions: []string{"v1", "v1beta1"},
-				},
+	ginkgo.Context("Running a conversion webhook service on a non-default port", func() {
+		servicePort = alternateServicePort
```

Comment:
I lose track of which closures get run when. I thought variable assignments had to be done in beforeEach blocks, but maybe I'm mistaken

Reply:
Dropped the contexts in the latest PR update.